### PR TITLE
[otbn] Add stall support to the ISS

### PR DESCRIPTION
**Pull request by rswarbrick**
With these patches, the ISS knows to stall for a cycle on LW and BN.LID. The first two patches in the queue are semi-unrelated cleanups, but they clash with the stalling patch, so it seemed easiest to push everything up together.

@GregAC: I think that LW and BN.LID are the only instructions that stall at the moment. Have I missed anything?

### CODE REVIEW

**Comment by rswarbrick** [comment]
Force-push fixes the text of a comment (I vacillated for a while on naming, and it seems I wrote that comment when wavering to the wrong side!)

File: hw/ip/otbn/data/insns.yml [hunk 1]
```diff
@@ -82,6 +82,11 @@ encoding-schemes: enc-schemes.yml
 #  straight-line: A boolean. If true, this instruction has no effect on control
 #                 flow. Optional, default true.
 #
+#  stall-cycles: A non-negative integer. If positive, this is the
+#                number of extra cycles that the instruction takes to
+#                execute (so an instruction takes 2 cycles if
+#                stall-cycles is 1). Optional, default 0.
```
Comment (imphil) [inline hw/ip/otbn/data/insns.yml:88]: I would find it more intuitive if we weren't specifying stall cycles, but cycles an instruction takes to execute. So stall-cycles -> cycles, with a default of 1. (In my mind, a stall is something blocking the pipeline. We don't have that, loads/stores are simply 2-cycle instructions. And there might be other instructions in the future with the same property, such as a multiplication which takes 2 cycles.)

**Review by imphil — COMMENTED** [review]
> I think that LW and BN.LID are the only instructions that stall at the moment. Have I missed anything?

Yes, that's correct (AFAIK).

File: hw/ip/otbn/data/insns.yml (hunk 1 above)
Comment (rswarbrick) [reply hw/ip/otbn/data/insns.yml:88]: Good point. For some reason I was trying to come up with something that defaulted to zero, but that's a bit silly! I'll change it.

File: hw/ip/otbn/data/insns.yml [hunk 2]
```diff
@@ -82,6 +82,9 @@ encoding-schemes: enc-schemes.yml
 #  straight-line: A boolean. If true, this instruction has no effect on control
 #                 flow. Optional, default true.
 #
+#  insn-cycles: A positive integer, giving the number of cycles that the
```
Comment (imphil) [inline hw/ip/otbn/data/insns.yml:85]:
```suggestion
#  cycles: A positive integer, giving the number of cycles that the
```

**Review by tomroberts-lowrisc — APPROVED** [review]
LGTM - agree its a bit weird to add the stall in the step function, but probably doesn't matter for now!

**Review by imphil — APPROVED** [review]

File: hw/ip/otbn/data/insns.yml (hunk 2 above)
Comment (rswarbrick) [reply hw/ip/otbn/data/insns.yml:85]: Drat! Clicked too soon. Will follow up with this.

**Comment by GregAC** [comment]
> @GregAC: I think that LW and BN.LID are the only instructions that stall at the moment. Have I missed anything?

That is correct

**Pull request merged**